client/lib/domains/registration/availability-messages.js

```diff
@@ -258,4 +258,149 @@ function getAvailabilityNotice( domain, error, site ) {
 	};
 }
 
-export { getAvailabilityNotice };
+function getAvailabilityNoticeForTransfers(
+	domain,
+	status,
+	site,
+	rootDomain,
+	mappableStatus,
+	goToMapDomainStep
+) {
+	let message,
+		severity = 'error';
+
+	const isSubDomain = rootDomain && rootDomain !== domain;
+
+	switch ( status ) {
+		case domainAvailability.MAPPABLE:
+		case domainAvailability.TLD_NOT_SUPPORTED:
+			if ( isSubDomain ) {
+				message = translate(
+					'Transferring a subdomain is not possible. You can try to {{button}}map it{{/button}} instead ' +
+						'or transfer {{strong}}the root domain{{/strong}}.',
+					{
+						args: { rootDomain },
+						components: {
+							strong: <strong />,
+							button: <button onClick={ goToMapDomainStep } />,
+						},
+					}
+				);
+			} else {
+				const tld = getTld( domain );
+				message = translate(
+					"We don't support transfers for domains ending with {{strong}}.%(tld)s{{/strong}}, " +
+						'but you can {{button}}map it{{/button}} instead.',
+					{
+						args: { tld },
+						components: {
+							strong: <strong />,
+							button: <button onClick={ goToMapDomainStep } />,
+						},
+					}
+				);
+				severity = 'info';
+			}
+			break;
+		case domainAvailability.UNKNOWN:
+			if ( domainAvailability.MAPPABLE === mappableStatus ) {
+				message = translate(
+					"{{strong}}%(domain)s{{/strong}} can't be transferred. " +
+						'You can {{button}}manually connect it{{/button}} if you still want to use it for your site.',
+					{
+						args: { domain },
+						components: {
+							strong: <strong />,
+							button: <button onClick={ goToMapDomainStep } />,
+						},
+					}
+				);
+				severity = 'info';
+				// The break is here intentionally so we can fallback to the default
+				break;
+			}
+		default:
+			( { message, severity } = getAvailabilityNotice( domain, status, site ) );
+	}
+
+	return {
+		message,
+		severity,
+	};
+}
+
+function getOfferAvailabilityForSearchResults(
```

Comment:
I thought we agreed that doing this juggling of boolean flags just doesn't scale - if you had a bug in this code, it seriously hard to spot it.
For example, I have a mapped subdomain and I've searched for it - I don't see the notice anymore that it's already mapped, instead I see the notice to map it again. What's worse, it seems to happen even in NUX, when I'm not logged in (as the user that owns that subdomain).
This different approach is also inconsistent with the other functions in this file.

Reply:
Hmm - I never mentioned but that PR goes with D10037-code

Reply:
Yup, I figured as much ;) But that still doesn't untangle the booleans inside. Check out how it looked like in version 1.0: https://github.com/Automattic/wp-calypso/commit/cabaf131c7bdfd58bcf0e9798f3439e06e484b34#diff-a36618f44b2f6aeae8145056394c517aL56 Looks kinda familiar? ;) And that was without transfers in and without subdomain support. And it did not work well... That's why I want to avoid coming back to that idea.